src/abstractions/software/vm-translator/command/index.js

```diff
@@ -0,0 +1,299 @@
+     * commands when the segment name is `static`
+     * @type {string}
+     */
+    this.stringArg = ''
+
+    /**
+     * Number of arguments (arg1 plus arg2). So for example if both arg1
+     * and arg2 are set, number of arguments is set to 2
+     * @type {number}
+     */
+    this.numberOfArgs = 0
+
+    // calculate and update number of arguments
+    this.updateNumberOfArgs()
+  }
+
+  /**
+   * Set arg1 of the HVM command, which can be one of the following:
+   * - segment name for push/pop commands
+   * - function name for function/call commands
+   * - label name for label and goto/if-goto commands
+   * @param {string} arg1 first argument of HVM command
+   * @throws {CommandException} if setting arg1 doesn't make sense
+   */
+  setArg1 (arg1) {
+    if (!doTypesMatch('local', arg1)) {
+      throw new CommandException(
+        "arg1 should be of type 'string', " +
+        ` but has type '${typeof arg1}'`)
+    }
+    if (this.commandType === COMMAND.RETURN) {
+      throw new CommandException(
+        'cannot set arg1 to a return command')
+    }
+    if (isArithmeticCommand(this.commandType)) {
+      throw new CommandException(
+        'cannot set arg1 to an arithmetic command')
+    }
+    if ([COMMAND.PUSH, COMMAND.POP].includes(this.commandType) &&
+      !isSegmentName(arg1)) {
+      throw new CommandException(
+        `invalid segment name: ${arg1} to a ${this.commandType} command`)
+    }
+    this.arg1 = arg1
+    this.updateNumberOfArgs()
+  }
+
+  /**
+   * Set arg2 of the HVM command, arg2 can be one of the following:
+   * - segment index for push/pop commands
+   * - number of arguments for call command
+   * - number of local variables for function command
+   * @param {number} arg2 arg2 of HVM command
+   * @throws {CommandException} if setting arg2 doesn't make sense
+   */
+  setArg2 (arg2) {
+    if (!doTypesMatch(2, arg2)) {
+      throw new CommandException(
+        "arg1 should be of type 'number', " +
+        ` but has type '${typeof arg2}'`)
+    }
+    if (!this.isArg2Relevant(this.commandType)) {
+      throw new CommandException(
+        `cannot set arg2 to the ${this.commandType} command`)
+    }
+    if ([COMMAND.POP, COMMAND.PUSH].includes(this.commandType)) {
+      if (arg2 < 0) {
+        throw new CommandException(`negative segment index: ${arg2}`)
+      }
+    }
+    this.arg2 = arg2
+    this.updateNumberOfArgs()
+  }
+
+  /**
+   * Additional info for the command, for example `className` for push/pull
+   * commands when the segment name is `static`
+   * @param {string} stringArg the stringArg of the HVM command
+   */
+  setStringArg (stringArg) {
+    this.stringArg = stringArg
+  }
+
+  /**
+   * Sets number of arguments to 2 if both arg1 and arg2
+   * are present, to 1 if either is missing, to 0 if both
+   * are not set
+   */
+  updateNumberOfArgs () {
+    const isArg0 = this.arg1 !== undefined
+    const isArg1 = this.arg2 !== undefined
+    if (isArg0 && isArg1) {
```

Comment:
Wondering what would be the trade of if we just use 
```if (this.arg1 && this.arg2) {```

I imagine it handles more cases, like `null` and empty strings. 
But yes, it fails when the argument is 0.

Reply:
yeah, you are right, it will fail with 0